### PR TITLE
VideoPress Onboarding: don't show site preview for VideoPress flow if a supported plan has not been purchased

### PR DESCRIPTION
**Pull request by jgcaruso**
#### Proposed Changes

* prevent iframe from loading the site preview if a plan has not been purchased, since the preview won't have the VideoMaker theme applied
* add a new site option to allow D97066-code to better detect the purchase so the theme is properly applied after plan purchase outside of the onboarding flow

Screenshot 2023-01-04 at 3 43 05 PM

#### Testing Instructions

* visit `/setup/videopress`
* Go through the flow up to the Checkout page, do not check out
* Press the `X` in the upper left to cancel
* Select either `Empty Cart` or `Leave items`
* You will be brought to the wpcom dashboard and will NOT see the site preview
* Press the `W` logo in the upper left
* You will be returned to the Launchpad
* Press Choose a Plan
* Select a plan
* Complete the purchase and skip the Plugins upgrade question (if you see it)
* Press the `W` logo again
* You should be returned to the Launchpad
* You will see the site preview, with the correct theme!

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/launchpad-site-preview.tsx [hunk 1]
```diff
@@ -16,15 +19,33 @@ const LaunchpadSitePreview = ( {
...
+		const productSlug = site?.plan?.product_slug;
+		const isVideoPressFlowWithUnsupportedPlan = ! planHasFeature(
+			productSlug as string,
+			FEATURE_VIDEO_UPLOADS
+		);
+
+		if ( isVideoPressFlowWithUnsupportedPlan ) {
+			previewUrl = null;
+			loadingMessage = translate(
+				'{{strong}}Site preview not available.{{/strong}} Plan purchase is required.',
```
Comment (roundhill) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/launchpad-site-preview.tsx:43]: I've seen other flows avoid `purchase` and use something like `upgrade` instead. Maybe `Plan upgrade is required?`

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/launchpad-site-preview.tsx [hunk 2]
```diff
@@ -1,8 +1,11 @@
+import { FEATURE_VIDEO_UPLOADS, planHasFeature } from '@automattic/calypso-products';
 import { DEVICE_TYPES } from '@automattic/components';
-import { NEWSLETTER_FLOW, VIDEOPRESS_FLOW } from '@automattic/onboarding';
+import { NEWSLETTER_FLOW } from '@automattic/onboarding';
 import { addQueryArgs } from '@wordpress/url';
 import { useTranslate } from 'i18n-calypso';
 import WebPreview from 'calypso/components/web-preview/component';
+import { useSite } from 'calypso/landing/stepper/hooks/use-site';
+import { isVideoPressFlow } from 'calypso/signup/utils';
```
Comment (roundhill) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/launchpad-site-preview.tsx:9]: Nice 👍

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/launchpad-site-preview.tsx (hunk 1 above)
Comment (roundhill) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/launchpad-site-preview.tsx]: How about `Plan upgrade is required`.?

**Review by roundhill — APPROVED** [review]
Tested great for me! had one copy thought.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7711610

Thank you @jgcaruso for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7711610).

**Pull request merged**